### ISSUE TITLE
/obs.html 

### ISSUE THREAD

**Issue by dr-code** [obsidian-html/obsidian-html#201]
Hi

First off thank you for this template, it really is awesome and ive been searching for a way to get my obsidian notes to publish without using Publish.

The issue I am having is with the /obs.html folder. It seems when I run localhost, the page is able to pick up where the css is located and formats accordingly. However when I transfer the output/html folder to a GitHub repo, it somehow can't find that same css due to the '/' in front of the obs.html. I have attached a picture of the issue.

Interestingly, when I manually remove the '/' from the index.html file from each of the css/js links it somehow works. But it is not realistic to manually remove each '/'.

I was wondering if you had any insight into this. 

thank you again

<img width="1800" alt="2022-03-30 Screen Shot 2022-03-30 at 8 05 05 AM" src="https://user-images.githubusercontent.com/8603561/160831184-ccc63d09-a506-4377-b655-8252504e3579.png">

**Comment by dwrolvink** [comment]
That is very weird... Does the obs.html folder exist in the root of your git repo? 
The absolute html path should just work when viewed via an http server, unless you host the website under a prefix like dc-code.github.io/**prefix/**, which you aren't.

Not sure why this isn't working.

It might be your browser. To exclude that possibility, can you look at https://obsidian-html.github.io/ ?
If that looks normal then we can exclude a lot of things.

This is also a github site, it has the obs.html folder directly in the git repo: 
https://github.com/obsidian-html/obsidian-html.github.io/tree/main/obs.html
With the full path: `<link rel="stylesheet" href="/obs.html/static/main.css" />` (view-source:https://obsidian-html.github.io/)

I can't check your code (https://github.com/dr-code?tab=repositories), can't look at the url either, I guess you removed the repo?

In any case, I've added the setting to create relative html links: https://obsidian-html.github.io/Configurations/Relative_path_html.html

Enabling that setting should do the same as what you did manually (you would get a path like ./obs.html/... instead of /obs.html/...)
But as the notes say, this will break the tab system and the graph view, so these will be disabled if you turn this setting on. You would get a site like this: https://obsidian-html.github.io/no_tabs/

Could you also give me the output of `pip show obsidianhtml` ?

**Comment by dwrolvink** [comment]
Just did a test, and relative_path_html actually **does** work with graph view and tabs, it just doesn't work without a webserver.
I've removed the auto-disable code, so you can just use relative_path_html: True with full functionality.

You can install the updated code via this method until a pip release is made: https://obsidian-html.github.io/Instructions/Run%20custom%20code.html
(I'm going to run some more tests, this might actually make the code a lot more simple in the long run)

Though we should really get to the bottom of this error because it doesn't make any sense.

**Comment by dr-code** [comment]
I tried that link in my current browser and it worked fine. 

I removed it so that I can try doing it all over again in case I messed up somewhere. I appreciate your response. In all honesty I am a bit of a noob when it comes to all of this code. I tried to install the updated code via the link you sent however I think that I needed to create the 'config.yaml' file first as the original instructions say. I may install it the original way and then add the `toggle relative_path_html` to the `config.yaml` file. 

I actually wanted it without the tabs anyways so that works out. 

here is my `pip show obsidianhtml`:
`Name: obsidianhtml
Version: 1.2.1
Summary: Converts Obsidian notes into proper markdown and HTML
Home-page: https://github.com/obsidian-html/obsidian-html
Author: https://github.com/dwrolvink
Author-email: dwrolvink@protonmail.com
License: GNU General Public License v3 or later (GPLv3+)
Location: /opt/homebrew/lib/python3.9/site-packages
Requires: beautifulsoup4, html5lib, markdown, obsidianhtml-md-mermaid-fork, pygments, python-frontmatter, regex, requests
Required-by: 
*****@*****-M1-Pro ~ % `

Let me give that `relative_path_html` a shot and ill get back to you. I appreciate your help. I also did not understand exactly how to automate it so that it goes straight to GitHub Pages. The workflow I want: I make changes in my obsidian notes, commit and push to GitHub and it shows up on the GitHub pages. 

Thanks

**Comment by dwrolvink** [comment]
np, this is my hobby, I'm happy to help you. It also shows me what parts are confusing/don't work well for beginners so that I can make the whole process easier, so we are really helping each other.

You should have a config.yaml already, otherwise the whole project will not work at all (you wouldn't get html code). You probably used the wrong path to the config.yml 

Are you on Windows or Linux btw? 

**Edit**: just created a bugfix release because someone found a breaking bug. So now you can just upgrade the package

``` bash
pip install obsidianhtml --upgrade
```

Then for a very basic config, take the code below and edit the paths to point to the correct locations. 
You can use an absolute path like e.g. `C:\users\dr-code\Documents\obsidian-vault\` or `/home/dr-code/Documents/obsidian-vault`. Save this as config.yml in a folder where you don't mind an output folder being created:

``` yaml
# The first folder that contains all obsidian files
# Use full path or relative path, but don't use ~/
obsidian_folder_path_str: '/path/to/your/vault' 

# The note that will be used as the index.html 
# should be in obsidian_folder_path_str
# Use full path or relative path, but don't use ~/
obsidian_entrypoint_path_str: '/path/to/your/vault/entrypoint.md'

toggles:
  relative_path_html: True
  no_tabs: True
```

Then open a terminal _in the same folder as where you just saved the config.yml_. Then run the following to convert your obsidian notes to html:
``` bash
obsidianhtml -i config.yml
```

The html will end up in your current folder under `output/html`.
Then start up a simple webserver (this is only for local testing, you can skip this when building your workflow)

``` bash
python -m http.server --directory output/html
```

Then open a new tab in your browser and go to `http://localhost:8000`
That should work.

If you get stuck anywhere post the output here and at what step you get which output and we can go through it step by step.

-------

I have a write-up on your exact usecase already (because that is my exact usecase for the documentation website). But the write-ups are a bit... they could use a rewrite let's say it like that. 
Especially if you are a beginner it would be nice to have some more explanations and stuff. But for reference, they are here: https://obsidian-html.github.io/MOCs/Automate%20website%20deployment.html#!using-a-git-repo-as-an-intermediate

If you let me know what OS you are on I can make an actual instruction on how to do that part if you are interested.

**Comment by dr-code** [comment]
I am using a MacBook. 

So something interesting when I make my config.yaml file and put the path, this is the error I get. 

```
---@--- PulmCrit % obsidianhtml -i config.yaml
> COPYING VAULT /Users/---/Documents/4. Career/4.8 Knowledge Bank/PulmCrit/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit TO /var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpblrh8qos
Traceback (most recent call last):
  File "/opt/homebrew/bin/obsidianhtml", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/__init__.py", line 490, in main
    tmpdir = CreateTemporaryCopy(source_folder_path=paths['obsidian_folder'], pb=pb)
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/lib.py", line 269, in CreateTemporaryCopy
    copy_tree(source_folder_path, tmpdir.name, preserve_times=1)
  File "/opt/homebrew/lib/python3.9/site-packages/setuptools/_distutils/dir_util.py", line 123, in copy_tree
    raise DistutilsFileError(
distutils.errors.DistutilsFileError: cannot copy tree '/Users/---/Documents/4. Career/4.8 Knowledge Bank/PulmCrit/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit': not a directory
```


If you look at the directory its trying to find, it seems like its repeating itself. So what I did was the following in my config.yaml file: 

```
# The first folder that contains all obsidian files
# Use full path or relative path, but don't use ~/
obsidian_folder_path_str: '' 

# The note that will be used as the index.html 
# should be in obsidian_folder_path_str
# Use full path or relative path, but don't use ~/
obsidian_entrypoint_path_str: 'entrypoint.md'

toggles:
  relative_path_html: True
  no_tabs: True
```

And that fixed the issue. 

### My Installation
So I was able to get the localhost up and running. Seems to be working fine. I guess I need help with how to get it to publish on GitHub and then use my own domain on it. I tried to follow the link you sent on automating it but truth be told I am not sure where to even put that code. And when I try to create a new repo with that folder I created with all the files, it doesn't automatically go to the index.html file because that is in output/html. I was getting some commit issues that I don't think I understand. So for now I deleted the repo. I am only running this on my localhost. Some guidance here would be greatly appreciate it. 

### A small aesthetic thing
I have attached a picture regarding this.
<img width="1800" alt="Screen Shot 2022-03-31 at 9 23 45 PM" src="https://user-images.githubusercontent.com/8603561/161176847-8a7227f1-43bd-41f8-acb6-1ea06aa0f16f.png">

Again I would like to thank you for your time and effort, and patience with a beginner like me.

**Comment by dwrolvink** [comment]
Now I'm curious what you had before you changed the path to an empty string :)

I think what you had was: `Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit` ?

That would be a relative path, so it would get added to whatever your working directory is, so if you are in the same folder, you would get:

**(workdir)**/(relative_path)   
**/Users/---/Documents/4. Career/4.8 Knowledge Bank/PulmCrit**/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit

changing your setting to 
``` yaml
obsidian_folder_path_str: '/Users/---/Documents/4. Career/4.8 Knowledge Bank/PulmCrit' 
```
Should fix that then, as it is an absolute path (notice the slash at the beginning, denoting an absolute path). This would then work from any working directory, but if you always run the command from that same folder then this is fine too.

Regarding the width of the page, adding the following code at the **bottom** of `/obs.html/static/main.css` would fix this:
``` css
body .single_tab_page {
    max-width: 100vw;
    width: 100vw;
    margin: 0rem;
}
```
If it does nothing, it might be a css caching issue that many browsers have, try then opening `http://localhost/obs.html/static/main.css` refreshing the page, and then refreshing `http://localhost/`.

My next post will explain how to set up the automation of pushing to github.

**Comment by dwrolvink** [comment]
Alright, I've never worked on a Mac before, but from what I know, it is very similar to linux, so I'll just give you the linux commands and hopefully that should all just work as-is.

A very long post, but it's just a lot of explanations, but the script you'll find at the bottom is very small and easy to understand if you can follow along with the (long) post above it. Once you have the script you can just run it every single time :)

It would be best to try each step separately, checking whether you get the result that is described, and then continue. 

Make sure you have one terminal open that you stay in for all the steps, and don't go changing folders unless the instructions tell you to. Open a separate terminal if you want to check different folders), to avoid running commands in the wrong directory or something like that, as that will give confusing errors.

# Paths
From what you've shared, I'm going to assume some paths that are of importance, change these if they are not correct (at least change `---` to your username I guess?)
``` bash
VAULT_PATH="/Users/---/Documents/4. Career/4.8 Knowledge Bank/PulmCrit"
GIT_FOLDER_PATH="/Users/---/Documents/git"

GIT_USER="dr-code"
REPO_NAME="dr-code.github.io"
REPO_PATH="$GIT_FOLDER_PATH/$REPO_NAME"

# Create the git folder if it does not exist
mkdir -p $GIT_FOLDER_PATH
```
At this point you should have an empty folder at /Users/---/Documents/git/ (or a non empty folder if you already had a dedicated git folder.

``` bash
# move into the git folder
cd $GIT_FOLDER_PATH
```

# The git repo
- Make an empty repo on github. Name it `<your_github_username>.github.io`. 
  - Edit: I see it already exists, this doesn't matter, if it is not empty, it will just give you different output in the check steps below 
- I assume you know how to do git clone and the likes

``` bash
git clone git@github.com:$GIT_USER/$REPO_NAME.git
```

You should get output like this:
``` plaintext
$ git clone git@github.com:$GIT_USER/$REPO_NAME.git                                                                                                                                                         
Cloning into 'empty'...
warning: You appear to have cloned an empty repository.
```
The last error will be absent if you have a readme.md or .gitignore in your repo.

``` bash
# move into the git folder
cd $REPO_NAME
```

What makes a git folder a git folder is the (hidden) `.git` sub directory. If you run `ls -la` it should show up:
``` plaintext
$ ls -la 
total 12
drwxr-xr-x  3 dorus dorus 4096 Apr  1 12:11 .
drwxr-xr-x 30 dorus dorus 4096 Apr  1 12:11 ..
drwxr-xr-x  7 dorus dorus 4096 Apr  1 12:13 .git
```

## Empty the git repo
Now what we want to do every time that we create your site, is to remove everything from this folder, **except** for the .git folder. Removing this folder will turn it into a normal folder, and then running a command like `git add .` will return an error like "current folder is not a git directory" or something similar.

``` bash
# Create a test file
touch test.txt

# Create the __src folder, we will put our custom css code in here at a later step for that tweak that you wanted
mkdir __src

# Check if it shows up
ls -la
```
You should now see 
``` plaintext
total 16
drwxr-xr-x  4 dorus dorus 4096 Apr  1 12:47 .
drwxr-xr-x 30 dorus dorus 4096 Apr  1 12:11 ..
drwxr-xr-x  7 dorus dorus 4096 Apr  1 12:13 .git
drwxr-xr-x  2 dorus dorus 4096 Apr  1 12:47 __src
-rw-r--r--  1 dorus dorus    0 Apr  1 12:16 test.txt
```
Now to remove everything except for the .git folder, the __src folder,  and the README.md (if you want to add it in the future)

``` bash
# This will find all folders/files directly in your current directory, excluding .git, the current dir itself ('.'), 
# and the readme; and print the output
find . -maxdepth 1 ! -path '.' ! -path "./.git/*" ! -name ".git" ! -name README.md ! -name __src -print0 | xargs -0 -I {} echo "{}"

# This should give you only 'test.txt', don't continue to the next command if it gives you back more!
# The next command is a recursive delete, if you do this in the wrong folder, you might loose more data than you intend to.

# This will remove everything that was found in the previous step
find . -maxdepth 1 ! -path '.' ! -path "./.git/*" ! -name ".git" ! -name README.md ! -name __src -print0 | xargs -0 -I {} /bin/rm -rf "{}"
```
Now if you do `ls -la` again, you should see that only test.txt was removed.

# Compile html in the correct location
Now we want to compile the html directly into the git folder.
Based on what you wrote before, I'm assuming your config.yml is in your vault folder.

``` bash
CONFIG_FILE_PATH="$VAULT_PATH/config.yml"
echo "$CONFIG_FILE_PATH"
```
Make sure the config.yml exists in the location that you get back. Otherwise set CONFIG_FILE_PATH to the correct path.

``` bash
# get the repo path for the next step
echo "$REPO_PATH"

# get the custom css path for the next step
CUSTOM_CSS_PATH="$REPO_PATH/__src/custom.css"
echo "$CUSTOM_CSS_PATH"
touch $CUSTOM_CSS_PATH
```

Open the file that you got back for CUSTOM_CSS_PATH with your favorite editor, and add in the following code:
``` css
body .single_tab_page {
    max-width: 100vw;
    width: 100vw;
    margin: 0rem;
}
```

Then, open the config.yml, and add the following settings (on top of the paths that we talked about before):
``` yaml
# Output path of HTML files
# This can be an absolute or a relative path (relative to the working directory when calling obsidianhtml)
# Use full path or relative path, but don't use ~/
html_output_folder_path_str: '<the path you got back for REPO_PATH>'

html_custom_inclusions:
  - '<the path you got back for CUSTOM_CSS_PATH>'

toggles:
  # This will skip emptying output folders, if you want to implement this yourself
  no_clean: True

  # Don't implements tabs
  no_tabs: True
```

This will make sure that 
- obsidianhtml puts the html files directly into your git folder
- it includes your custom css code
- it does not empty the html output directory by itself (the issue you probably had earlier)
- it disables the tab system

You can add in the relative_path_html here too, if you still need it for some weird reason.

See bottom of this post for the complete config.yml that I used.

> See https://obsidian-html.github.io/Configurations/Configuration%20Options.html for a full list of settings.

Now we can compile the html
``` bash
obsidianhtml -i $CONFIG_FILE_PATH

# check if index.html is present in the git repo folder
ls | grep "index.html"
```
This should return one line.

Okay, now you can do the regular checks, like starting the webserver, going to localhost in the browser, and check if everything looks alright.

# Git commit & push
``` bash
# Add the new files to the list to be committed
git add . --all

# If you havent done this yet, set this information, or git will complain
git config --global user.name "FIRST_NAME LAST_NAME"
git config --global user.email "MY_NAME@example.com"

# Make a new commit (a unit of change)
git commit -m "first commit"

# Push the code to your repo
git push
```

Now you can open https://github.com/dr-code/dr-code.github.io and see if the commit is there.

Once the proceeding works, it is all downhill from there, because now we can just create a script to do all this for us.

# Building the script
``` bash
# Set location for the new script (you can change this to whatever you like)
SCRIPT_PATH= "/Users/---/Documents/publish_vault.sh"

# Make the new  script file
touch $SCRIPT_PATH

# Look up some values to use in the script
echo $CONFIG_FILE_PATH
echo $REPO_PATH
```
Then open the new script, and add in the following code:

``` bash
#!/usr/bin/env bash

# Config
CONFIG_FILE_PATH="Change this to what you got for CONFIG_FILE_PATH"
REPO_PATH="Change this to what you got for REPO_PATH"

# Move into the correct folder
cd $REPO_PATH

# Empty folder
echo "Will delete previous output:"
find . -maxdepth 1 ! -path '.' ! -path "./.git/*" ! -name ".git" ! -name README.md ! -name __src -print0 | xargs -0 -I {} echo "{}"
echo "Continue? (y/n)"
read yn
if [ "$yn" != "y" ]; then
    echo "Aborted"
    exit 0
fi

find . -maxdepth 1 ! -path '.' ! -path "./.git/*" ! -name ".git" ! -name README.md ! -name __src -print0 | xargs -0 -I {} /bin/rm -rf "{}"

# Compile site
obsidianhtml -i $CONFIG_FILE_PATH

git add . --all
git commit -m "autopush"
git push
```

For the script that I use regularly, I've removed the check at the beginning, and the git commands, because I always check locally first, and then just git commit/push using vs code, so I have the following code:

``` bash
#!/usr/bin/env bash

# Config
CONFIG_FILE_PATH="/home/dorus/git/obsidian-html/ci/test_vault/config.yml" 
REPO_PATH="/home/dorus/git/empty"

# Move into the correct folder
cd $REPO_PATH

# Empty folder
find . -maxdepth 1 ! -path '.' ! -path "./.git/*" ! -name ".git" ! -name README.md ! -name __src -print0 | xargs -0 -I {} /bin/rm -rf "{}"

# Compile site
obsidianhtml -i $CONFIG_FILE_PATH
```
# Run the script
First, we need to grant execute permissions to the script file
``` bash
chmod +x $SCRIPT_PATH
```

Now we can (finally) run it:
``` bash
$SCRIPT_PATH
```

And that's it :)

-----------

My config.yml:

``` yaml
# The first folder that contains all obsidian files
# Can be absent when toggles/compile_md == False
# On linux, use fullpath instead of ~/
obsidian_folder_path_str: '/home/dorus/git/obsidian-html/ci/test_vault'

# The note that will be used as the index.html 
# should be in obsidian_folder_path_str
# Can be absent when toggles/compile_md == False
# On linux, use fullpath instead of ~/
obsidian_entrypoint_path_str: '/home/dorus/git/obsidian-html/ci/test_vault/entrypoint.md'

# Output path of HTML files
# This can be an absolute or a relative path (relative to the working directory when calling obsidianhtml)
# Use full path or relative path, but don't use ~/
html_output_folder_path_str: '/home/dorus/git/empty'

# Add in custom html inclusions
html_custom_inclusions:
  - '/home/dorus/git/empty/__src/custom.css'

toggles:
  # This will skip emptying output folders, if you want to implement this yourself
  no_clean: True

  # Don't implements tabs
  no_tabs: True
```

**Comment by dwrolvink** [comment]
Btw I see that you are trying out Logseq now, don't feel obliged to continue with obsidianhtml if that works better for you.

The post that I wrote will be useful for our documentation at some point, so it is not wasted effort from my side if you decide to switch :+1:

**Comment by dr-code** [comment]
Hello again! 

Thank you for the detailed steps. 
The only reason I was contemplating Logseq was because I couldn’t find a good way to get these notes from obsidian. I have over 300 files on obsidian so it would be a huge undertaking if I switched so I would rather not. Either way not wasted effort. Albeit it may be because I try to follow along and sometimes can't.

Where I am stuck:
Everything was fine and dandy until the step of 

> Now we can compile the html
```
 obsidianhtml -i $CONFIG_FILE_PATH
 # check if index.html is present in the git repo folder
 ls | grep "index.html"
```
> This should return one line.

Just so we are on the same page I will write all my paths:
```
VAULT_PATH="/Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit"
GIT_FOLDER_PATH="/Users/rutvik/Documents/1.\ Personal\ Development/1.1\ Projects/git"

GIT_USER="dr-code"
REPO_NAME="dr-code.github.io"
REPO_PATH="/Users/rutvik/Documents/1.\ Personal\ Development/1.1\ Projects/git/dr-code.github.io"

CONFIG_FILE_PATH="/Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/config.yaml"
CUSTOM_CSS_PATH="/Users/rutvik/Documents/1.\ Personal\ Development/1.1\ Projects/git/dr-code.github.io/__src/custom.css"
```

The following is my config.yaml file:
```
# The first folder that contains all obsidian files
# Use full path or relative path, but don't use ~/
obsidian_folder_path_str: '/Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit' 

# The note that will be used as the index.html 
# should be in obsidian_folder_path_str
# Use full path or relative path, but don't use ~/
obsidian_entrypoint_path_str: '/Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/entrypoint.md'

html_output_folder_path_str: '/Users/rutvik/Documents/1.\ Personal\ Development/1.1\ Projects/git/dr-code.github.io'
html_custom_inclusions:
  - '/Users/rutvik/Documents/1.\ Personal\ Development/1.1\ Projects/git/dr-code.github.io/__src/custom.css'

site_name: 'Pulmonary Critical Care Notes'

toggles:
  relative_path_html: True
  no_tabs: True
  features:
    graph:
      enabled: True          # Include code to build the graph view per page (default: True)
      coalesce_force: '-100' # How close the nodes will be to eachother.
    create_index_from_dir_structure:
      enabled: True
      exclude_subfolders:
        - ".git"
        - "__src"
        - "md"
        - "obs.html"
      exclude_files:
        - "index.html"
        - "favicon.ico"
        - "not_created.html"
    # Show backlinks at the bottom of each note
    backlinks:
      enabled: True
  no_clean: True
```

So when I run `obsidianhtml -i /Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/config.yaml`

I get the following error:
```
rutvik@Rutviks-M1-Pro PulmCrit % obsidianhtml -i /Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/config.yaml
> COPYING VAULT /Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit TO /var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmp1iacuul0
Traceback (most recent call last):
  File "/opt/homebrew/bin/obsidianhtml", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/__init__.py", line 490, in main
    tmpdir = CreateTemporaryCopy(source_folder_path=paths['obsidian_folder'], pb=pb)
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/lib.py", line 269, in CreateTemporaryCopy
    copy_tree(source_folder_path, tmpdir.name, preserve_times=1)
  File "/opt/homebrew/lib/python3.9/site-packages/setuptools/_distutils/dir_util.py", line 123, in copy_tree
    raise DistutilsFileError(
distutils.errors.DistutilsFileError: cannot copy tree '/Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit': not a directory
```

In the config.yaml file when I edit the obsidian_folder_path_str and obsidian_entrypoint_path_str: to just (Just to test it out because of the previous path error I was getting)

> `obsidian_folder_path_str: ''`
>  `obsidian_entrypoint_path_str: 'entrypoint.md'`

I get the following error:
```
rutvik@Rutviks-M1-Pro PulmCrit % obsidianhtml -i /Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/config.yaml
> COPYING VAULT /Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit TO /var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmp2g3d736a
< COPYING VAULT: Done
> CREATING OUTPUT FOLDERS
Traceback (most recent call last):
  File "/opt/homebrew/bin/obsidianhtml", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/__init__.py", line 566, in main
    raise DuplicateFileNameInRoot(f"Two or more files with the name \"{path.name}\" exist in the root folder. See {str(path)} and {pb.files[path.name]['fullpath']}.")
TypeError: 'OH_File' object is not subscriptable
```

Not sure where I am going wrong. 
Thanks

**Comment by dwrolvink** [comment]
I think I know what is going wrong. Bash will give you back backslashes in front of spaces, but in yaml, a backslash is just a backslash. So if you go back to your original setup and remove all the backslashes from your paths in the yaml then it should work.

As far as I can tell, you can leave the backslashes in all the other places.

``` yaml
# The first folder that contains all obsidian files
# Use full path or relative path, but don't use ~/
obsidian_folder_path_str: '/Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit' 

# The note that will be used as the index.html 
# should be in obsidian_folder_path_str
# Use full path or relative path, but don't use ~/
obsidian_entrypoint_path_str: '/Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit/entrypoint.md'

html_output_folder_path_str: '/Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io'
html_custom_inclusions:
  - '/Users/rutvik/Documents/1. Personal\ Development/1.1 Projects/git/dr-code.github.io/__src/custom.css'
```

Though then we come to the second error. I made a mistake in the error, so that's nice. But it complains that you have multiple files in the vault that have the same name.

Just fixed this right now. If you do the following, you will get the new code and it should tell you which two files share the same name. 
``` bash
pip uninstall obsidianhtml
pip install git+https://github.com/obsidian-html/obsidian-html
```
If one of the two can be ignored, then you can ignore it by adding the file (or entire folder if it can be ignored) to the `exclude-subfolders` setting in your config.yml. See also https://obsidian-html.github.io/Configurations/Configuration%20Options.html#!exclude-subfolders

So if there is a file named `/Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit/random/file.md` then you would add this to your config.yml:

```
exclude_subfolders:
  - ".obsidian"
  - ".trash"
  - ".DS_Store"
  - "random/file.md"
```

Let me know if that gets you a bit further

**Comment by dr-code** [comment]
I uninstalled and re-installed based on your instructions.
I corrected my config.yaml file as above.

I went to run `obsidianhtml -i /Users/rutvik/Documents/4.\ Career/4.8\ Knowledge\ Bank/PulmCrit/config.yaml `

This is the error I get:

```
rutvik@Rutviks-M1-Pro PulmCrit % obsidianhtml -i config.yaml
> COPYING VAULT /Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit TO /var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u
< COPYING VAULT: Done
> CREATING OUTPUT FOLDERS
Traceback (most recent call last):
  File "/opt/homebrew/bin/obsidianhtml", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/lib/python3.9/site-packages/obsidianhtml/__init__.py", line 568, in main
    raise DuplicateFileNameInRoot(f"Two or more files with the name \"{path.name}\" exist in the root folder. See {str(path)} and {pb.files[path.name].path[path_type]['file_absolute_path']}.")
obsidianhtml.lib.DuplicateFileNameInRoot: Two or more files with the name "index.html" exist in the root folder. See /private/var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u/output/html/obs.html/tags/index.html and /private/var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u/output/html/index.html.
```

I tried to go locate that folder - and even the terminal stated that the folder doesn't exist. 

```
rutvik@Rutviks-M1-Pro PulmCrit % cd /private/var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u
cd: no such file or directory: /Users/rutvik/private/var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u
```

EDIT:
So then I tried to do:
```
exclude_subfolders:
  - ".obsidian"
  - ".trash"
  - ".DS_Store"
  - "/private/var/folders/rg/fwn3m75x30x4q0579df86rtm0000gn/T/tmpc1df9g_u/output/html"
```
in my config.yaml file and it returns the same error.

**Comment by dwrolvink** [comment]
Ahhh you've created the output folder in your obsidian folder in an earlier step.
Go to `/Users/rutvik/Documents/4. Career/4.8 Knowledge Bank/PulmCrit` there is probably a folder there called `output`, this is generated by obsidianhtml when you were using your previous settings. Remove that and run again.

It is correct that the folder does not exist, it is a temporary folder created to avoid bugs ruining your vault. The folder is removed when obsidianhtml exits, but the contents of the folder are the same as your obsidian folder.

Good idea to exclude the folder, but every temporary folder gets a unique name, so that won't work :stuck_out_tongue:

**Comment by dr-code** [comment]
Ah that makes sense. 
So that fixed that problem.

I think I am having an issue with the custom.css file
![Screen Shot 2022-04-01 at 2 28 13 PM](https://user-images.githubusercontent.com/8603561/161321553-3fe815ae-20cd-4e1d-9635-765b6936a599.png)

When I take a look at the index.html file I see where the error is:

```
<!DOCTYPE html>

<html lang="en">

        <head>
                <!-- Page information -->
                <meta charset="UTF-8" />
                <meta name="node_id" content="index">
                <meta name="viewport" content="width=device-width, initial-scale=1.0" />
                <link rel="shortcut icon" href="/favicon.ico" />

                <!-- Set title -->
                <title>Pulmonary Critical Care Notes</title>

                <!-- Includes -->
                <link rel="stylesheet" href="./obs.html/static/mermaid.css" />
                <link rel="stylesheet" href="./obs.html/static/main.css" />
                <script src="./obs.html/static/mermaid.min.js"></script> 
                /Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io/__src/custom.css
<link rel="stylesheet" href="./obs.html/static/graph.css" />
<script src="https://d3js.org/d3.v4.min.js"></script>
<script src="./obs.html/static/dirtree.js" /></script>
```

You notice its not listed as a `<script src="..."</script>`

This is the section in my config.yaml file that I put it for the html_custom_inclusions:
```
html_output_folder_path_str: '/Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io'
html_custom_inclusions:
  - '/Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io/__src/custom.css'
```

When I remove the custom css lines that issue goes away but of course that right sided bar comes back.

**Comment by dwrolvink** [comment]
ah right, sorry, I forgot that you have to add in html code there
so:

``` yaml
html_custom_inclusions:
  - '<link rel="stylesheet" href="./__src/custom.css" />'
```

Though that won't work with the relative paths, because if you go into a folder then the link will not work anymore. So if you still can't turn relative paths off at this point (for a still undisclosed reason), then there are a couple of different options:

- export the html template, add the custom css inline, and configure it in your config.yml
- add some code to your script to inject css into main.css after compilation

I guess the last one is the easiest for now, so let's try that (unless you can turn off relative path, did you try that again?)

### inject css after compilation
First, remove html_custom_inclusions from your config.yaml, we won't need that anymore

Run this after the obsidianhtml -i ... step:
``` bash
CUSTOM_CSS_PATH='/Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io/__src/custom.css'
MAIN_CSS_PATH='/Users/rutvik/Documents/1. Personal Development/1.1 Projects/git/dr-code.github.io/obs.html/static/main.css'

echo '\n' >> $MAIN_CSS_PATH
cat $CUSTOM_CSS_PATH >> $MAIN_CSS_PATH
```

That should inject your custom css into the main file, so it is added correctly everywhere.

**Comment by dr-code** [comment]
Tried turning off the relative path again, and it worked. 

The custom.css worked on my localhost websever

I was able to push everything to GitHub (https://github.com/dr-code/dr-code.github.io)

My html_custom_inclusions: in my config.yaml file was as follows:
```
html_custom_inclusions:
  - '<link rel="stylesheet" href="/__src/custom.css"/>'
```

When I try to go to the GitHub Pages link (https://dr-code.github.io/); you can see that the sidebar is still there. Then I go to inspect element to see if its recognizing the custom.css file and it shows "An error occurred trying to load that resource"

The following is the <head> element in my index.html code:
```
<!DOCTYPE html>

<html lang="en">

        <head>
                <!-- Page information -->
                <meta charset="UTF-8" />
                <meta name="node_id" content="index">
                <meta name="viewport" content="width=device-width, initial-scale=1.0" />
                <link rel="shortcut icon" href="/favicon.ico" />

                <!-- Set title -->
                <title>Pulmonary Critical Care Notes</title>

                <!-- Includes -->
                <link rel="stylesheet" href="/obs.html/static/mermaid.css" />
                <link rel="stylesheet" href="/obs.html/static/main.css" />
                <script src="/obs.html/static/mermaid.min.js"></script> 
                <link rel="stylesheet" href="/__src/custom.css"/>
<link rel="stylesheet" href="/obs.html/static/graph.css" />
<script src="https://d3js.org/d3.v4.min.js"></script>
<script src="/obs.html/static/dirtree.js" /></script>


                <!-- Onload tweaks -->
                <script>
                        // add prefix to hash to avoid scrolling
                        if (1 == 0){
                                if (window.location.hash != '' && window.location.hash.substr(1)[0] != '!'){
                                        window.location.hash = '!' + window.location.hash.substr(1);
                                }
                        }
                </script>
        </head>
```

Coincidentally if I click on the  `<link rel="stylesheet" href="/obs.html/static/main.css" />`; I am able to see the main.css file.

**Comment by dwrolvink** [comment]
try putting an empty file in your repo called .nojekyll
apparently, folders that start with underscores are not copied over to github pages.

another solution would be to just rename __src to src (and then update all the links)

https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/

edit: works for me (github pages updates about 5 minutes after pushing if you hadn't noticed yet)

**Comment by dwrolvink** [comment]
congrats! :confetti_ball:  
I see it worked :)

**Comment by dr-code** [comment]
Yes it did!! 

Thank you so much! 

Would love to see where this project goes!
